### PR TITLE
Dashboard Settings

### PR DESCRIPTION
**Pull request by vargaat**
refs:
affects: Student, Teacher, Parent
release note: Added a dedicated dashboard settings menu.

test plan:
- Profile menu shouldn't have dashboard related switches.
- Dark mode switch should still be available in the parent app.
- Dashboard layout selection button should be replaced by a settings button.
- Tapping the settings button opens the dashboard settings screen.
- Teacher app shouldn't have a switch for grade display.
- Added Lato semi-bold font to the app so semi-bold fonts are now correctly rendered semi-bold instead of bold.

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

<table>
<tr><th>iPhone</th><th>iPad</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Parent Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: CoreTests/FontExtensionTests/testFonts</summary>


    Crash: CoreTester (16085) implicit closure #47 in FontExtensionTests.testFonts()
</details>
<details><summary>:x: XCTest failed: CoreTests/UIFontExtensionsTests/testScaledNamedFont</summary>


    Crash: CoreTester (16334) implicit closure #1 in UIFontExtensionsTests.testScaledNamedFont()
</details>


#### Affected Apps: Student, Teacher, Parent

#### Release Note: 
Added a dedicated dashboard settings menu.
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

File: Core/Core/Dashboard/View/DashboardCardView.swift [hunk 1]
```diff
@@ -103,14 +106,39 @@ public struct DashboardCardView: View {
...
+                    popoverController.sourceRect = CGRect(x: 26, y: 35, width: 0, height: 0)
+                }
+
+                env.router.show(
+                    container,
+                    from: controller,
+                    options: .modal(.popover),
+                    analyticsRoute: "/dashboard/settings"
+                )
+            } label: {
```
Comment (szabolcs-instructure) [inline Core/Core/Dashboard/View/DashboardCardView.swift:136]: I think creating view controllers, view models, interactors are out of the scope of the view's responsibility. A view should signal to its' view model that a button has been pressed and the view model should delegate the creation/routing further up.

File: Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    // MARK: - Outputs
+    @Published public private(set) var layout: DashboardLayout
+    public let isGradesSwitchVisible: Bool
+    public let isColorOverlaySwitchVisible: Bool
+    public let popoverSize: CGSize
+
+    // MARK: - Inputs
+    public let setGridLayout = PassthroughSubject<Void, Never>()
+    public let setListLayout = PassthroughSubject<Void, Never>()
```
Comment (szabolcs-instructure) [inline Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift:34]: We could use a typed subject like `PassthroughSubject<DashboardLayout, Never>()` so we don't need an extra property and mapping.

File: Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            .removeDuplicates()
+            .assign(to: &$layout)
+    }
+
+    private func bindUserInputsToInteractor() {
+        let gridPublisher = setGridLayout.map { DashboardLayout.grid }
+        let listPublisher = setListLayout.map { DashboardLayout.list }
+        Publishers.Merge(gridPublisher, listPublisher)
+            .subscribe(interactor.layout)
+            .store(in: &subscriptions)
```
Comment (szabolcs-instructure) [inline Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift:67]: Having one publisher would mean we could wire it up directly to `interactor.layout`.

**Review by szabolcs-instructure — CHANGES_REQUESTED** [review]
I had some comments about simplifying publishers and moving view controller creation from view.

**Review by idevainstr — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/79920680/200826694-d717b2c5-6d6c-4d70-b616-b0e9a591be04.MP4

Getting instant crashes on cog button tap on iPadOS 16.1, in both the Teacher and Student apps.

IMG_0057

Landscape layout safe area is miscolored on iOS 14.8.1.

File: Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift (hunk 2 above)
Comment (vargaat) [reply Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift]: I don't even know why I did this way, nice catch!

File: Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift (hunk 3 above)
Comment (vargaat) [reply Core/Core/Dashboard/ViewModel/Settings/DashboardSettingsViewModel.swift]: Did that way!

**Review by szabolcs-instructure — APPROVED** [review]

**Review by idevainstr — APPROVED** [review]
QA +1

**Review by davidvarga93 — APPROVED** [review]
QA + 1 
iOS 16.2b, iPad OS 14.6

**Pull request merged**